### DISCUSSION TITLE
Non-blocking usage of Kubernetes Client using Vert.x Http Client

### DISCUSSION THREAD

**Discussion by firdy94** [fabric8io/kubernetes-client#5067] [Q&A]
Hi, I have been following this issue: https://github.com/fabric8io/kubernetes-client/issues/2764 
Saw that the vert.x httpclient implementation is now used for the kubernetes client and the latest quarkus version has migrated to use it. 

Have been trying to find usages of the kubernetes client with vertx http client implementation being used in a non-blocking way but have not come across anything like that. 

Had a query regarding how I can configure the vertx http client to be used by the kubernetes client and use the client to achieve non-blocking communication with the kube api?

**Comment by shawkins** [comment]
> Had a query regarding how I can configure the vertx http client to be used by the kubernetes client and use the client to achieve non-blocking communication with the kube api?

The jdk, vertx, and jetty httpclients all offer better internal non-blocking support that utilize a shared core thread pool, but none of them change the overall DSL exposed by the fabric8 client.  For most of the DSL there isn't a direct option for non-blocking usage.  We're talked about ways to better expose this, but haven't committed to a path forward.
